LambdaEngine/Source/ECS/EntitySubscriber.cpp

```diff
@@ -19,6 +19,38 @@ namespace LambdaEngine
         ComponentAccesses = componentAccesses;
     }
 
+    EntitySubscriptionRegistration::EntitySubscriptionRegistration(const TArray<ComponentAccess>& componentAccesses, const TArray<IComponentGroup*>& componentGroups, const TArray<const ComponentType*>& excludedComponentTypes, IDVector* pSubscriber, std::function<void(Entity)>onEntityAdded, std::function<void(Entity)>onEntityRemoval)
+        :EntitySubscriptionRegistration(componentAccesses, {}, pSubscriber, onEntityAdded, onEntityRemoval)
+    {
+        // Filter the component accesses present in excluded list
+        TArray<ComponentAccess> componentAccessesFiltered;
+        bool exclude = false;
+
+        for (auto component : ComponentAccesses)
+        {
+            // Look for component access in excluded list
+            for (const auto* excludedCompType : excludedComponentTypes)
+            {
+                if (component.TID == excludedCompType)
+                {
+                    exclude = true;
+                    break;
+                }
+            }
+
+            // Include component access if not in excluded list
+            if (!exclude) 
+            {
+                componentAccessesFiltered.PushBack(component);
+            }
+
+            exclude = false;
```

Comment:
The algorithm is good, but as the linter suggests, you can use an std algorithm instead, such as:

```suggestion

            if (std::none_of(excludedComponentTypes.Begin(), excludedComponentTypes.End()
                    (const auto* pExcludedType) { pExcludedType == component.TID; }))
            {
                    componentAccessesFiltered.PushBack(component);
            }
```

And then the `exclude` variable can be removed as well.